### PR TITLE
bugfix: vendor latest libnetwork for connect panic

### LINKED ISSUE #1550: [bug] connect network panic
### Ⅰ. Issue Description

after restart the pouch daemon, connect a container to a network failed. The pouchd panic

### Ⅱ. Describe what happened

1. run a container

```
root@ubuntu-1:~/pouch# pouch run -d -t reg.docker.alibaba-inc.com/busybox:latest top
c8e969acb6cacb9e74119f4788fcaa2f77b7c2a1c7ad804af07390124bbc7035
root@ubuntu-1:~/pouch# pouch ps
Name     ID       Status         Created         Image                                       Runtime
c8e969   c8e969   Up 2 seconds   2 seconds ago   reg.docker.alibaba-inc.com/busybox:latest   runc
```

2. create a network

```
root@ubuntu-1:~/pouch# pouch network create net1 -d bridge
net1: a1c2be6ea445b3e8a3a59ddac3d30aa4d72f2649e20fa31149c6ba5b75c1e751
```

3. restart the pouchd

4. connect the container to the network

```
root@ubuntu-1:~/pouch# pouch network connect net1 c8e969
Error: Post http://d/v1.24/networks/net1/connect: EOF
```

pouch daemon log

```
INFO[2018-06-19T15:59:12.294025096+08:00] Calling POST /v1.24/networks/net1/connect, client @ 
DEBU[2018-06-19T15:59:12.294974667+08:00] create endpoint for container [c8e969acb6cacb9e74119f4788fcaa2f77b7c2a1c7ad804af07390124bbc7035] on network [net1] 
DEBU[2018-06-19T15:59:12.295471631+08:00] Assigning addresses for endpoint c8e969ac's interface on network net1 
DEBU[2018-06-19T15:59:12.295984634+08:00] RequestAddress(LocalDefault/172.18.0.0/16, <nil>, map[]) 
DEBU[2018-06-19T15:59:12.301527689+08:00] Assigning addresses for endpoint c8e969ac's interface on network net1 
2018/06/19 15:59:12 http: panic serving @: assignment to entry in nil map
goroutine 173 [running]:
net/http.(*conn).serve.func1(0xc420a64460)
	/usr/local/go/src/net/http/server.go:1697 +0xd0
panic(0x1ab5520, 0x1d7fb20)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/docker/libnetwork.JoinOptionPriority.func1(0xc420139e40)
	/tmp/pouchbuild/src/github.com/docker/libnetwork/endpoint.go:943 +0xf7
github.com/docker/libnetwork.(*endpoint).processOptions(0xc420139e40, 0xc42000fd28, 0x1, 0x1)
	/tmp/pouchbuild/src/github.com/docker/libnetwork/endpoint.go:381 +0x9d
github.com/docker/libnetwork.(*endpoint).sbJoin(0xc420139a20, 0xc42007cb40, 0xc42000fd28, 0x1, 0x1, 0x0, 0x0)
	/tmp/pouchbuild/src/github.com/docker/libnetwork/endpoint.go:448 +0x38a
github.com/docker/libnetwork.(*endpoint).Join(0xc420139a20, 0x267ab80, 0xc42007cb40, 0xc42000fd28, 0x1, 0x1, 0x0, 0x0)
	/tmp/pouchbuild/src/github.com/docker/libnetwork/endpoint.go:414 +0xcc
github.com/alibaba/pouch/daemon/mgr.(*NetworkManager).EndpointCreate(0xc420354a00, 0x2672f60, 0xc4200f5780, 0xc420139760, 0x0, 0x0, 0x0, 0x0)
	/tmp/pouchbuild/src/github.com/alibaba/pouch/daemon/mgr/network.go:325 +0x67d
github.com/alibaba/pouch/daemon/mgr.(*ContainerManager).connectToNetwork(0xc4200703f0, 0x2672f60, 0xc4200f5780, 0xc4203368c0, 0xc4202e8915, 0x4, 0xc42011ab40, 0x0, 0xc42031d9e8)
	/tmp/pouchbuild/src/github.com/alibaba/pouch/daemon/mgr/container.go:1588 +0x184
github.com/alibaba/pouch/daemon/mgr.(*ContainerManager).Connect(0xc4200703f0, 0x2672f60, 0xc4200f5780, 0xc420a0ef98, 0x6, 0xc4202e8915, 0x4, 0xc42011ab40, 0xc420516b80, 0x4)
	/tmp/pouchbuild/src/github.com/alibaba/pouch/daemon/mgr/container.go:1427 +0x179
github.com/alibaba/pouch/apis/server.(*Server).connectToNetwork(0xc4204fc570, 0x2672f60, 0xc4200f5780, 0x2671d20, 0xc420445b20, 0xc420a5ea00, 0x3, 0x3)
	/tmp/pouchbuild/src/github.com/alibaba/pouch/apis/server/network_bridge.go:95 +0x2f4
github.com/alibaba/pouch/apis/server.(*Server).(github.com/alibaba/pouch/apis/server.connectToNetwork)-fm(0x2672f60, 0xc4200f5780, 0x2671d20, 0xc420445b20, 0xc420a5ea00, 0xc420a5ea00, 0xc42089dad0)
	/tmp/pouchbuild/src/github.com/alibaba/pouch/apis/server/router.go:79 +0x5c
github.com/alibaba/pouch/apis/server.filter.func1(0x2671d20, 0xc420445b20, 0xc420a5ea00)
	/tmp/pouchbuild/src/github.com/alibaba/pouch/apis/server/router.go:184 +0x327
net/http.HandlerFunc.ServeHTTP(0xc420945230, 0x2671d20, 0xc420445b20, 0xc420a5ea00)
	/usr/local/go/src/net/http/server.go:1918 +0x44
github.com/alibaba/pouch/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc4208cc1e0, 0x2671d20, 0xc420445b20, 0xc420a5ea00)
	/tmp/pouchbuild/src/github.com/alibaba/pouch/vendor/github.com/gorilla/mux/mux.go:133 +0xed
net/http.serverHandler.ServeHTTP(0xc4200fd860, 0x2671d20, 0xc420445b20, 0xc420a5e200)
	/usr/local/go/src/net/http/server.go:2619 +0xb4
net/http.(*conn).serve(0xc420a64460, 0x2672f60, 0xc4200f5640)
	/usr/local/go/src/net/http/server.go:1801 +0x71d
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2720 +0x288
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1.
2.
3.

### Ⅵ. Environment:

- pouch version (use `pouch version`):

```
GoVersion:       go1.9.5
KernelVersion:   4.4.0-116-generic
Os:              linux
Version:         0.5.0
APIVersion:      1.24
Arch:            amd64
BuildTime:       2018-06-19T15:56:04+08:00
GitCommit:       0.4.0-448-g1ac9207
```

- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by shaloulcy**
It seems a bug of libnetwork. When libnetwork restores the sandbox, it forgets to initialize the epPriority map. So the Join panic. The latest alibaba/libnetwork has fixed the bug. We will vendor the latest libnetwork and add some test cases
    
    https://github.com/alibaba/libnetwork/commit/655f08f8be32ff847d0554ef0128a3de80665cd6

Signed-off-by: Eric Li <lcy041536@gmail.com>

### Ⅰ. Describe what this PR did

It seems a bug of libnetwork. When libnetwork restores the sandbox, it forgets to initialize the epPriority map. So the Join panic. The latest alibaba/libnetwork has fixed the bug. We will vendor the latest libnetwork and add some test cases

### Ⅱ. Does this pull request fix one issue?

fixes #1550

### CODE REVIEW

**Comment by allencloud** [comment]
Do we need to add integration test to cover issue #1550 ? @shaloulcy

**Comment by shaloulcy** [comment]
@allencloud I have added integration test. PTAL

File: daemon/mgr/network.go [hunk 1]
```diff
@@ -727,7 +727,5 @@ func joinOptions(endpoint *types.Endpoint) ([]libnetwork.EndpointOption, error)
 	var joinOptions []libnetwork.EndpointOption
 	// TODO: parse endpoint's links
 
-	// set priority option
-	joinOptions = append(joinOptions, libnetwork.JoinOptionPriority(nil, endpoint.Priority))
```
Comment (rudyfly) [inline daemon/mgr/network.go]: we will use this feature, can you fix this bug in our libnetwork(https://github.com/alibaba/libnetwork) ?

**Comment by rudyfly** [comment]
LGTM

**Pull request merged**